src/qemu/qemu_command.c

```diff
@@ -10937,11 +10937,21 @@ qemuBuildSerialChrDeviceProps(const virDomainDef *def,
         return NULL;
     }
 
-    if (virJSONValueObjectAdd(&props,
-                              "s:driver", virDomainChrSerialTargetModelTypeToString(serial->targetModel),
-                              "s:chardev", chardev,
-                              "s:id", serial->info.alias,
-                              NULL) < 0)
+    if (serial->targetModel == VIR_DOMAIN_CHR_SERIAL_TARGET_MODEL_ISA_SERIAL &&
+                    serial->target.port != -1) {
+            if (virJSONValueObjectCreate(&props,
+                                 "s:driver", virDomainChrSerialTargetModelTypeToString(serial->targetModel),
+                                 "s:chardev", chardev,
+                                 "s:id", serial->info.alias,
+                                 "i:index", serial->target.port,
+                                 NULL) < 0)
+                    return NULL;
+    }
+    else if (virJSONValueObjectCreate(&props,
+                                 "s:driver", virDomainChrSerialTargetModelTypeToString(serial->targetModel),
```

Comment:
Also are changes for qemuBuildSerialChrDeviceStr already there is master, to update traget port for domain xml.
case VIR_DOMAIN_CHR_SERIAL_TARGET_MODEL_ISA_SERIAL:
+        if (serial->target.port != -1)
+            virBufferAsprintf(&cmd, ",index=%d", serial->target.port);
+        break;

Reply:
not that change is not there. It was not followed up.